src/utils/Recorder.ts

```diff
@@ -0,0 +1,83 @@
+import { MockedRequest } from './handlers/requestHandler'
+import { fetch } from '../context/fetch'
+
+async function createFunctionFromRequest(
+  request: MockedRequest,
+  response: Response,
+) {
+  const lines = []
+  const composeLines = []
+  lines.push(
+    `rest.${request.method.toLowerCase()}('${
+      request.url
+    }',function(req,res,ctx){`,
+  )
+
+  composeLines.push(`ctx.status(${response.status})`)
+  const body = await response.text()
+
+  const hasJsonContent = response.headers?.get('content-type')?.includes('json')
+  if (hasJsonContent) {
+    composeLines.push(`ctx.json(${body})`)
+  } else {
+    composeLines.push(`ctx.body(${body})`)
+  }
+
+  response.headers.forEach((value, key) => {
+    composeLines.push(`ctx.set('${key}', '${value}')`)
```

Comment:
How would this handle a header with multiple values?

Reply:
`ctx.set` will append the value to the Header so if one value for the same key already exist the will be merged